akka-docs/src/main/paradox/general/stream/stream-design.md

```diff
@@ -45,6 +45,35 @@ All stream Processors produced by the default materialization of Akka Streams ar
 
 This means that `Sink.asPublisher(true)` (for enabling fan-out support) must be used where broadcast behavior is needed for interoperation with other Reactive Streams implementations.
 
+### Rationale and benefits from Sink/Source/Flow not directly extending Reactive Streams interfaces
+
+An often overlooked fact about [Reactive Streams](https://github.com/reactive-streams/reactive-streams-jvm/) is that they are a [Service Provider Interface](https://en.m.wikipedia.org/wiki/Service_provider_interface).
+This is confirmed even in the [earliest discussions](https://github.com/reactive-streams/reactive-streams-jvm/pull/25) about the specification dating back to 2014.
+Akka Streams was designed along with the development of Reactive Streams, with heavy involvement of the Akka team, as such, we understand the purpose of these interfaces very well.
+
+In the earliest days of the specification even the spec's types attempted to hide Publisher, Subscriber and the other types.
+Sadly the types would leak regardless in the back then considered API, [thus the API(!) was removed and the SPI types are all that remained][9].
+
+Nowadays you see some implementations of Reactive Streams claim that their direct extending of these types is a benefit for some reason.
+This is not correct, as such was not, and is not the goal of the Reactive Streams interfaces.
+It is rather a misunderstanding of what these types are -- strictly the inter-op interfaces that Reactive Streams libraries agree to understand and "speak" (a protocol).
+
+For reference, other libraries like RxJava 2.0 as well as Reactor do directly extend these the Reactive Stream types in their user-facing types, while Akka Streams remains true to the Reactive Streams's design principles by hiding them as an application developer programming interface -- which is why Sink does not extend Subscriber, yet can *expose* it on demand.
+
+`Source`, `Sink` and `Flow` types are part of Akka Streams, and have their direct counter parts in Reactive Streams (being, respectively, `Publisher`, `Subscriber` and `Processor`), and their purpose is to provide the fluent DSL,
+as well as be "factories" the actual runtime -- often also called "blueprints". Akka Streams operate on a [lifted representation](https://doc.akka.io/docs/akka/2.5.6/scala/general/stream/stream-design.html) of the computing graph,
+which these types allow, which then is materialized and executed in accordance to Reactive Streams rules.
+
```

Comment:
here I think we should say something about how easy it is to integrate Akka Streams with other RS libraries, by using the `asPublisher` and `asSubscriber`

Reply:
right, fixed